abaverify/main.py

```diff
@@ -452,7 +452,21 @@ def _runAssertionsOnResults(self, jobName):
                 # Loop through values if there are more than one
                 if hasattr(r['computedValue'], '__iter__'):
                     for i in range(0, len(r['computedValue'])):
-                        self.assertAlmostEqual(r['computedValue'][i], r['referenceValue'][i], delta=r['tolerance'][i])
+                        computed_val = r['computedValue'][i]
+                        reference_val = r['referenceValue'][i]
+                        delta = r['tolerance'][i]
+                        if isinstance(computed_val, tuple):
+                            # --- when there exists a tuple as a reference val then all other results and deltas
+                            # --- should also be tuples
+                            self.assertEqual(len(computed_val), len(reference_val),
+                                             "Specified reference value should be same length as Computed value")
+                            self.assertEqual(len(computed_val), len(delta),
```

Comment:
I would like to see the tolerance have the option to be a single value that's applied to all computed y's (seems like a more likely use-case to me). If you want to keep the option to specify the tolerance as a list of tuples as it is currently, that's fine with me. Also, please refactor 'delta' to 'tolerance' so that line 469 reads: self.assertAlmostEqual(computed_val, reference_val, delta=tolerance) 

or similar. I'm suggesting this to keep the naming consistent.

Reply:
Newest commit allows both. If the tolerance is specified as a single tuple it as the tolerance used against all computed vals. Otherwise the corresponding tolerance (by common index) is used